Add a mention of our nightly builds to our `README.md`. (#4158)
carbon-language/carbon-lang: README.md

## Getting started

To try out Carbon immediately in your browser, you can use the Carbon explorer

> to interpret Carbon code and print its output on:

Reviewer: You might consider putting a little more emphasis on "demo interpreter for Carbon", like above, rather than just "Carbon".

Author: Done I think, PTAL.

print its output. You can try it out immediately at
[compiler-explorer.com](http://carbon.compiler-explorer.com/).

Carbon is an early, experimental project, and so we only have very experimental

> nightly releases of the Carbon toolchain available to download, and only on

Reviewer: FWIW, this feels a bit odd coming after the "explorer" advice above. [I'd lean towards removing the explorer device, or moving it below]

Author: I'd like to keep opening with the compiler-explorer link that folks can just dive into.

But I agree we need a transition in the text... I've tried to reword both sections a bit to emphasize that the above is just about trying stuff out directly in the browser. I've also removed the explorer reference from a local build below to focus in on the toolchain where we would potentially be interested in contributions.

PR description also updated a bit to reflect the changes here.

Reviewer: NB, I think you might've forgotten to push the commit for this.

Author: I... think it is pushed? Maybe it wasn't as much of an adjustment as you were looking for?

limited platforms. If you are using a recent Ubuntu Linux or similar (Debian,
WSL, etc.), you can try these out by going to our
[releases](https://github.com/carbon-language/carbon-lang/releases) page and
download the latest nightly toolchain tar file:

> `carbon_toolchain-v0.0.0-0.nightly.YYYY.MM.DD.tar.gz`. Then you can try it out:

Reviewer: ```suggestion
`carbon_toolchain-0.0.0-0.nightly.YYYY.MM.DD.tar.gz`. Then you can try it out:
```

```shell
# A variable with the specific nightly version:

> VERSION="v0.0.0-0.nightly.YYYY.MM.DD"

Reviewer: ```suggestion
VERSION="0.0.0-0.nightly.YYYY.MM.DD"
```

# Unpack the toolchain:
tar -xvf carbon_toolchain-${VERSION}.tar.gz

# Create a simple Carbon source file:

> echo "fn Run() { Print(42); }" > forty_two.carbon

Reviewer: ```suggestion
echo "fn Run() { Core.Print(42); }" > forty_two.carbon
```

# Compile to an object file:
./carbon_toolchain-${VERSION}/bin/carbon compile \
  --output=forty_two.o forty_two.carbon

# Link to an executable:

> ./carbon_toolchain-${VERSION}/bin/carbon link \

Reviewer: The latter instructions include clang, libc++, and lld. Do we need any of that in order to link?

Author: I don't *think* so. Definitely don't need clang or lld.

I'll try this out with a fresh Ubuntu cloud image just to see what if anything might be needed there.

@KateGregory -- anything we should include for WSL?

Author: I've added the minimal necessary `sudo apt install` based on my testing right before linking (it isn't needed until linking).

  --output=forty_two forty_two.o

# Run it:
./forty_two
```

However, the toolchain is still very early and many things don't yet work.

> Please hold off on filing lots of bugs, we know many parts of this don't work

Reviewer: ```suggestion
Please hold off on filing lots of bugs: we know many parts of this don't work
```

yet or may not work on all systems. We expect to have releases that are much
more robust and reliable that you can try out when we reach our
[0.1 milestone](/docs/project/milestones.md#milestone-01-a-minimum-viable-product-mvp-for-evaluation).

> If you do want to build Carbon's toolchain yourself and maybe contribute fixes

Reviewer: How much value do we get from providing both the release instruction and the `apt install` instruction here? Would it be worth removing the latter?

Author: I think it's nice to keep the source build snippets for now. I'm still interested in nudging folks interested to consider contributing.

I did prune the explorer part out as that's not really relevant for contributing at this point.

./carbon_toolchain-${VERSION}/bin/carbon compile \
  --output=forty_two.o forty_two.carbon

> # Install minimal system libraries used for linking, note that installing `gcc`

Reviewer: ```suggestion
# Install minimal system libraries used for linking. Note that installing `gcc`
```

> # or `g++` for compiling C/C++ code with GCC will also be sufficient, this is
> # just the specific system libraries Carbon linking currently still uses.

Reviewer: ```suggestion
# or `g++` for compiling C/C++ code with GCC will also be sufficient, this is
# just the specific system library Carbon linking currently uses.
```

```suggestion
# or `g++` for compiling C/C++ code with GCC will also be sufficient, these are
# just the specific system libraries Carbon linking still uses.
```
